### ISSUE TITLE
[lsp] fatal error

### ISSUE THREAD

**Issue by rgrinberg** [ocaml/merlin#1036]
I'm marking this as LSP, but I'm not sure if this error is related to LSP at all. So far I don't have a small reproduction case, but this exception occurs fairly regularly:

```
Fatal error: exception Syntaxerr.Error(_)
Raised at file "src/ocaml/preprocess/parser_raw.mly", line 498, characters 4-69
Called from file "src/ocaml/preprocess/parser_raw.mly", line 3325, characters 21-53
Called from file "src/utils/menhirLib.ml", line 1885, characters 18-44
Called from file "src/utils/std.ml", line 93, characters 17-25
Called from file "src/kernel/mreader_recover.ml", line 210, characters 10-57
Called from file "src/kernel/mreader_recover.ml", line 248, characters 28-38
Called from file "src/kernel/mreader_parser.ml", line 130, characters 21-35
Called from file "src/kernel/mreader_parser.ml", line 175, characters 20-48
Called from file "src/kernel/mreader_parser.ml", line 190, characters 6-42
Called from file "src/ocaml/utils/misc.ml", line 45, characters 8-15
Re-raised at file "src/ocaml/utils/misc.ml", line 57, characters 10-24
Called from file "src/kernel/mreader_parser.ml", line 204, characters 20-56
Called from file "src/kernel/mreader.ml", line 49, characters 15-77
Called from file "src/kernel/mpipeline.ml", line 103, characters 19-62
Called from file "camlinternalLazy.ml", line 29, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 36, characters 4-11
Re-raised at file "src/kernel/mpipeline.ml", line 15, characters 34-49
Called from file "camlinternalLazy.ml", line 29, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 36, characters 4-11
Called from file "camlinternalLazy.ml", line 29, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 36, characters 4-11
Re-raised at file "src/kernel/mpipeline.ml", line 15, characters 34-49
Called from file "camlinternalLazy.ml", line 29, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 36, characters 4-11
Called from file "camlinternalLazy.ml", line 29, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 36, characters 4-11
Re-raised at file "src/kernel/mpipeline.ml", line 15, characters 34-49
Called from file "camlinternalLazy.ml", line 29, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 36, characters 4-11
Called from file "src/kernel/mpipeline.ml" (inlined), line 78, characters 21-30
Called from file "src/frontend/query_commands.ml", line 64, characters 14-45
Called from file "src/frontend/lsp/ocamlmerlin_lsp.ml", line 145, characters 15-70
Called from file "src/frontend/lsp/ocamlmerlin_lsp.ml", line 690, characters 4-28
Called from file "src/utils/std.ml", line 679, characters 8-12
Re-raised at file "src/utils/std.ml", line 681, characters 30-39
Called from file "src/utils/std.ml", line 679, characters 8-12
Re-raised at file "src/utils/std.ml", line 681, characters 30-39
Called from file "src/lsp/rpc.ml", line 348, characters 21-25
Called from file "src/lsp/rpc.ml", line 391, characters 8-810
Called from file "src/frontend/lsp/ocamlmerlin_lsp.ml", line 725, characters 2-95
Called from file "src/utils/logger.ml", line 120, characters 8-12
Re-raised at file "src/utils/logger.ml", line 122, characters 53-64
Called from file "src/utils/logger.ml", line 148, characters 10-34
Re-raised at file "src/utils/logger.ml", line 150, characters 35-46
Called from file "src/frontend/lsp/ocamlmerlin_lsp.ml", line 749, characters 2-74
```

**Comment by trefis** [comment]
Seems like the same issue as #1032 (even though the backtraces are different).

**Comment by trefis** [comment]
Should have been fixed by #1037.
